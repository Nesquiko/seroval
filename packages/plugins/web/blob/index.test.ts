import { describe, it, expect } from 'vitest';
import {
  crossSerializeAsync,
  deserialize,
  fromCrossJSON,
  fromJSON,
  serializeAsync,
  toCrossJSONAsync,
  toJSONAsync,
} from 'seroval';
import BlobPlugin from '.';

const EXAMPLE = new Blob(['Hello World'], {
  type: 'text/plain',
});
describe('Blob', () => {
  describe('serializeAsync', () => {
    it('supports Blob', async () => {
      const result = await serializeAsync(EXAMPLE, {
        plugins: [BlobPlugin],
      });
      expect(result).toMatchSnapshot();
      const back = deserialize<typeof EXAMPLE>(result);
      expect(back).toBeInstanceOf(Blob);
      expect(await back.text()).toBe(await EXAMPLE.text());
      expect(back.type).toBe(EXAMPLE.type);
    });
  });
  describe('toJSONAsync', () => {
    it('supports Blob', async () => {
      const result = await toJSONAsync(EXAMPLE, {
        plugins: [BlobPlugin],
      });
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromJSON<typeof EXAMPLE>(result, {
        plugins: [BlobPlugin],
      });
      expect(back).toBeInstanceOf(Blob);
      expect(await back.text()).toBe(await EXAMPLE.text());
      expect(back.type).toBe(EXAMPLE.type);
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports Blob', async () => {
      const result = await crossSerializeAsync(EXAMPLE, {
        plugins: [BlobPlugin],
      });
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports Blob', async () => {
        const result = await crossSerializeAsync(Promise.resolve(EXAMPLE), {
          plugins: [BlobPlugin],
          scopeId: 'example',
        });
        expect(result).toMatchSnapshot();
      });
    });
  });
  describe('toCrossJSONAsync', () => {
    it('supports Blob', async () => {
      const result = await toCrossJSONAsync(EXAMPLE, {
        plugins: [BlobPlugin],
      });
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromCrossJSON<typeof EXAMPLE>(result, {
        plugins: [BlobPlugin],
        refs: new Map(),
      });
      expect(back).toBeInstanceOf(Blob);
      expect(await back.text()).toBe(await EXAMPLE.text());
      expect(back.type).toBe(EXAMPLE.type);
    });
  });
});
