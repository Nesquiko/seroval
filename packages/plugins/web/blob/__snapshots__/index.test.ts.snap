// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Blob > crossSerializeAsync > scoped > supports Blob 1`] = `"($R=>$R[0]=Promise.resolve($R[1]=new Blob([$R[2]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100]).buffer],{type:"text/plain"})))($R["example"])"`;

exports[`Blob > crossSerializeAsync > supports Blob 1`] = `"$R[0]=new Blob([$R[1]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100]).buffer],{type:"text/plain"})"`;

exports[`Blob > serializeAsync > supports Blob 1`] = `"new Blob([new Uint8Array([72,101,108,108,111,32,87,111,114,108,100]).buffer],{type:"text/plain"})"`;

exports[`Blob > toCrossJSONAsync > supports Blob 1`] = `"{"t":40,"i":0,"s":{"type":{"t":1,"s":"text/plain"},"buffer":{"t":21,"i":1,"s":[72,101,108,108,111,32,87,111,114,108,100]}},"c":"seroval-plugins/web/url"}"`;

exports[`Blob > toJSONAsync > supports Blob 1`] = `"{"t":{"t":40,"i":0,"s":{"type":{"t":1,"s":"text/plain"},"buffer":{"t":21,"i":1,"s":[72,101,108,108,111,32,87,111,114,108,100]}},"c":"seroval-plugins/web/url"},"f":16383,"m":[]}"`;
